system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Translation;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\App;
+use Locale;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+class LocalizationFinder extends BaseCommand
+{
+    protected $group       = 'Translation';
+    protected $name        = 'lang:find';
+    protected $description = 'Find and save available phrases to translate';
+    protected $usage       = 'lang:find [options]';
+    protected $arguments   = [];
+    protected $options     = [
+        '--locale'   => 'Apply changes to the selected locale (en, ru, etc...).',
+        '--dir'      => 'Directory for searching for translations (in relation to the APPPATH).',
+        '--show-new' => 'See only new translations as table.',
+        '--verbose'  => 'Output detailed information.',
+    ];
+
+    /**
+     * Flag for output detailed information
+     */
+    private bool $verbose = false;
+
+    private string $languagePath;
+
+    public function run(array $params)
```

Comment:
This method is too long, so difficult to understand.
Refactor with extracting private methods if you can.